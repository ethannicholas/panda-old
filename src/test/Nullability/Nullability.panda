method nullableString():String? {
    if Random.bit()
        return "string!"
    else
        return null
}

var s := nullableString()
if s != null
    Console.writeLine(s)
if s != null & s.length > 0
    Console.writeLine(s)
if s != null | (s != null & s.length > 0)
    Console.writeLine(s) -- FIXME TEST ERROR: shouldn't be getting an error here
Console.writeLine(s)
Console.writeLine(s.length)

s := nullableString()
assert s != null
Console.writeLine(s)