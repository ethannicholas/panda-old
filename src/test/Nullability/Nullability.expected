error: Nullability.panda:14:23: cannot use possibly-null value as non-nullable type panda.core.String
error: Nullability.panda:15:19: cannot use possibly-null value as non-nullable type panda.core.String
error: Nullability.panda:16:19: cannot use possibly-null value as non-nullable type panda.core.String
3 errors
